packages/office-ui-fabric-react/src/components/GroupedList/GroupedListV2.base.tsx

```diff
@@ -0,0 +1,619 @@
+
+    return (
+      <GroupItem
+        render={onRenderHeader}
+        defaultRender={renderGroupHeader}
+        item={item}
+        selection={selection}
+        eventGroup={events.current}
+        props={headerProps}
+      />
+    );
+  };
+
+  const renderShowAll = (item: IShowAllGroupedItem, flattenedIndex: number): React.ReactNode => {
+    const group = item.group;
+    const groupShowAllProps = {
+      ...groupProps!.showAllProps,
+      ...getDividerProps(group, flattenedIndex),
+      key: group.key ? `${group.key}-show-all` : undefined,
+    };
+
+    return onRenderShowAll(groupShowAllProps, renderGroupShowAll);
+  };
+
+  const renderFooter = (item: IFooterGroupedItem, flattenedIndex: number): React.ReactNode => {
+    const group = item.group;
+    const groupFooterProps = {
+      ...groupProps!.footerProps,
+      ...getDividerProps(group, flattenedIndex),
+      key: group.key ? `${group.key}-footer` : undefined,
+    };
+
+    return onRenderFooter(groupFooterProps, renderGroupFooter);
+  };
+
+  const renderItem = (item: IGroupedItem, flattenedIndex: number): React.ReactNode => {
+    if (item.type === 'header') {
+      return renderHeader(item, flattenedIndex);
+    } else if (item.type === 'showAll') {
+      return renderShowAll(item, flattenedIndex);
+    } else if (item.type === 'footer') {
+      return renderFooter(item, flattenedIndex);
+    } else {
+      const level = item.group.level ? item.group.level + 1 : 1;
+      return onRenderCell(level, item.item, item.itemIndex ?? flattenedIndex, item.group);
+    }
+  };
+
+  return (
+    <FocusZone
+      direction={FocusZoneDirection.vertical}
+      data-automationid="GroupedList"
+      data-is-scrollable="false"
+      role="presentation"
+      {...focusZoneProps}
+      shouldEnterInnerZone={shouldEnterInnerZone}
+      className={css(classNames.root, focusZoneProps.className)}
+    >
+      <List
+        ref={listRef}
+        role={role}
+        items={listView}
+        // eslint-disable-next-line
+        onRenderCellConditional={renderItem}
+        usePageCache={usePageCache}
+        onShouldVirtualize={onShouldVirtualize}
+        getPageSpecification={getPageSpecification}
+        version={version}
+        getKey={getKey}
+        {...rootListProps}
+      />
+    </FocusZone>
+  );
+};
+
+interface IGroupItemProps<T> {
+  props: T;
+  render: IRenderFunction<T>;
+  defaultRender: (props?: T) => JSX.Element | null;
+  item: any;
+  selection?: ISelection;
+  eventGroup?: EventGroup;
+}
+
+const GroupItem = <T,>({
+  render,
+  defaultRender,
+  item,
+  selection,
+  eventGroup,
+  props,
+}: React.PropsWithChildren<IGroupItemProps<T>>): React.ReactElement | null => {
+  const group = item.group;
+
+  const isSelected = useIsGroupSelected(group.startIndex, group.count, selection, eventGroup);
+  const mergedProps = {
+    ...props,
+    isSelected: isSelected,
```

Comment:
_nit:_ This can just be

```suggestion
    isSelected,
```